### PR TITLE
[rllib] Use batch.count in async samples optimizer

### PR DESCRIPTION
**Pull request by ericl**
## What do these changes do?

Using the actual batch size reduces the risk of mis-accounting. Here, we under-counted samples since in truncate_episodes mode we were doubling the batch size by accident in policy_evaluator.

## Related issue number

https://github.com/ray-project/ray/issues/2439

### CODE REVIEW

File: python/ray/rllib/optimizers/async_samples_optimizer.py [hunk 1]
```diff
@@ -226,16 +225,21 @@ def _step(self):
...
 
         with self.timers["sample_processing"]:
-            for ev, sample_batch in self.sample_tasks.completed():
-                self._check_not_multiagent(sample_batch)
-                sample_timesteps += self.sample_batch_size
+            completed = list(self.sample_tasks.completed())
+
+            # Bulk fetch all the batch counts for each batch
+            for count in ray.get([c[1][1] for c in completed]):
+                completed[1][1] = count
```
[the remark this answers is not in the document]
Comment (ericl) [reply python/ray/rllib/optimizers/async_samples_optimizer.py]: Oops, forgot to push the commit that fixes this.

**Review by ericl — COMMENTED** [review]
Fixed

**Comment by michaelschaarschmidt** [comment]
Just wondering re removing: 

-        "max_weight_sync_delay": 400,

any specific reason as it seemingly doesn't relate to issue?

File: python/ray/rllib/agents/dqn/apex.py [hunk 2]
```diff
@@ -23,7 +23,6 @@
         "learning_starts": 50000,
         "train_batch_size": 512,
         "sample_batch_size": 50,
-        "max_weight_sync_delay": 400,
```
Comment (richardliaw) [inline python/ray/rllib/agents/dqn/apex.py]: Good catch @michaelschaarschmidt; seems to be an accidental remove.

Comment (richardliaw) [reply python/ray/rllib/agents/dqn/apex.py]: @michaelschaarschmidt actually this was already provided in L15, so this is a duplicate 

(sorry for the previous message)

**Review by richardliaw — APPROVED** [review]

File: python/ray/rllib/agents/dqn/apex.py (hunk 2 above)
Comment (michaelschaarschmidt) [reply python/ray/rllib/agents/dqn/apex.py]: oops, thanks!

**Comment by richardliaw** [comment]
Jenkins tests are failing

**Pull request merged**